File: blockchain-info-aggregation-stream-parent/blockchain-info-aggregation-stream-adapter/src/main/java/info/blockchain/aggregation/adapter/websocket/server/WebSocketServerConfiguration.java
package info.blockchain.aggregation.adapter.websocket.server;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.Ordered;
import org.springframework.web.reactive.HandlerMapping;
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping;
import org.springframework.web.reactive.socket.WebSocketHandler;
import org.springframework.web.reactive.socket.server.support.WebSocketHandlerAdapter;

import static java.util.Collections.singletonMap;

@Configuration
public class WebSocketServerConfiguration {

    @Autowired
    @Qualifier("aggregate-handler")
    private WebSocketHandler webSocketHandler;


    @Bean
    public HandlerMapping handlerMapping() {
        SimpleUrlHandlerMapping simpleUrlHandlerMapping = new SimpleUrlHandlerMapping();
        simpleUrlHandlerMapping.setOrder(Ordered.HIGHEST_PRECEDENCE);
        simpleUrlHandlerMapping.setUrlMap(singletonMap("/aggregate", webSocketHandler));
        return simpleUrlHandlerMapping;
    }


    @Bean
    public WebSocketHandlerAdapter webSocketHandlerAdapter() {
        return new WebSocketHandlerAdapter();
    }

}

File: blockchain-info-aggregation-stream-parent/blockchain-info-aggregation-stream-adapter/src/main/java/info/blockchain/aggregation/adapter/websocket/server/WebSocketServerConfiguration.kt
package info.blockchain.aggregation.adapter.websocket.server

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.Ordered
import org.springframework.web.reactive.HandlerMapping
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping
import org.springframework.web.reactive.socket.WebSocketHandler
import org.springframework.web.reactive.socket.server.support.WebSocketHandlerAdapter

import java.util.Collections.singletonMap

@Configuration
class WebSocketServerConfiguration {

    @Autowired
    @Qualifier("aggregate-handler")
    private val webSocketHandler: WebSocketHandler? = null


    @Bean
    fun handlerMapping(): HandlerMapping {
        val simpleUrlHandlerMapping = SimpleUrlHandlerMapping()
        simpleUrlHandlerMapping.order = Ordered.HIGHEST_PRECEDENCE
        simpleUrlHandlerMapping.urlMap = singletonMap<String, WebSocketHandler>("/aggregate", webSocketHandler)
        return simpleUrlHandlerMapping
    }


    @Bean
    fun webSocketHandlerAdapter(): WebSocketHandlerAdapter {
        return WebSocketHandlerAdapter()
    }

}
